ebuild-writing/variables/text.xml

```diff
@@ -588,5 +588,54 @@ when joining paths. For example: <c>${D%/}/</c>, <c>${ED%/}/</c>,
 </body>
 </section>
 
+<section>
+<title>Use of constant-value variables in ebuilds</title>
+<body>
+
+<p>
+Variables frequently have significant value in ebuilds, making it
+possible to avoid unnecessary repetitions and make maintenance easier.
+However, references to constant-value variables should be used with care
+as their proactive use can harm readability and increase maintenance
+burden (e.g. when renaming the package). In particular, using variables
+whose values are not strictly guaranteed to match the expected string
+should be avoided.
+</p>
+
+<p>The examples of beneficial constant-value variable references are:</p>
+<ul>
+  <li>
+    using <c>${PV}</c> and the variables derived from it (<c>${P}</c>,
+    <c>${MY_P}</c>…) in <c>SRC_URI</c> and <c>S</c> to avoid having
+    to update those variables for every version bump (presuming that
+    <c>${PV}</c> is used to indicate the upstream version);
+  </li>
+
+  <li>
+    using <c>${PF}</c> in <c>--docdir</c> path — this is a canonical
+    Gentoo path that is always required to match <c>${PF}</c>.
+  </li>
+</ul>
+
+<p>The examples of bad constant-value variable references are:</p>
+<ul>
+  <li>
+    using <c>${PN}</c> in <c>HOMEPAGE</c>, <c>EGIT_REPO_URI</c>,
+    the domain or path prefix part of <c>SRC_URI</c> — it breaks URL
+    parsing in editors and terminals, and makes it hard to copy the link
+    for external use (e.g. while reviewing via gitweb),
```

Comment:
There are 11264 ebuilds that contain `${PN}` in `SRC_URI`. Unless we convince people to fix them first, there's little value in putting this in devmanual.

Per https://devmanual.gentoo.org/ebuild-writing/variables/index.html, developers are advised to not use variables inside `HOMEPAGE`, so that's  covered.

Reply:
I disagree about this hunk too. IMHO the general guideline from above ("references to constant-value variables should be used with care") is enough, and we should avoid too much micro-management here, like specific rules for `SRC_URI` or `${PN}`.

Reply:
Examples are not micro-management. Examples tip you what the formal rules actually mean. Would it be fine if I killed the part about `SRC_URI`?